posts/parent-class/index.qmd

```diff
@@ -0,0 +1,50 @@
+
+A peculiarity of R is the availability and centrality of `data.frame`s in the base language, whereas you need extra libraries for a similar functionality in most other languages (e.g., pandas in Python).
+
+`data.frame` is one of the first "complex" (in the sense of [non-atomic](https://adv-r.hadley.nz/vectors-chap.html#atomic-vectors)) object most R learners will be exposed to and will develop a familiarity with. A good way to leverage this familiarity is to make your subclass a thin wrapper around `data.frame`s.
+
+This means that not only users will be able to get started with your package faster because of this familiarity, but you will also immediately benefit for the huge ecosystem of functions and packages working on data.frames, such as the [tidyverse](https://tidyverse.org). If you want some examples, this is what we did in the [linelist](https://github.com/epiverse-trace/linelist), [pavo](https://github.com/rmaia/pavo), [scoringutils](https://github.com/epiforecasts/scoringutils), and [vaccineff](https://github.com/epiverse-trace/vaccineff) R packages.
+
+In some cases, the output is too complex to fit into a `data.frame`. Even in this situation, I would recommend inheriting from existing, well-established, classes for the same two reasons: familiarity and ecosystem. For example, for the [serofoi](https://github.com/epiverse-trace/serofoi) R package, [we have made the decision to inherit from `stanfit` objects, rather than a custom structure](https://github.com/epiverse-trace/serofoi/pull/117).
+
+## Rely on parent methods as much as possible
+
+A follow up recommendation from inheriting from standard classes is to leverage their methods wherever possible.
+
+[One of the first](https://github.com/epiverse-trace/linelist/pull/60) [changes I made](https://github.com/epiverse-trace/linelist/pull/61) when becoming maintainer of the [linelist](https://github.com/epiverse-trace/linelist) package was to remove the `rename.linelist()` and `select.linelist()` methods. Indeed, they were, or could easily be, behaving identically as the parent `rename.data.frame()` and `select.data.frame()` methods. Rather than burdening the codebase and maintenance with an extra unnecessary method, it is much simpler and more robust to rely on the well-tested parent method. In fact, the [dplyr documentation](https://dplyr.tidyverse.org/reference/dplyr_extending.html) explicitly recommends only writing methods for a couple of standard functions (including `[.subclass()` and `names<-.subclass()`), which will enable the use of parent methods directly, rather than writing custom methods for each dplyr functions.
+
+Similarly, many developers have the reflex to write a custom `print.subclass()` method as part of the method implementation. While it may be justified in some cases, it is also sometimes unnecessary extra fluff. My recommendation would be to evaluate carefully what benefit the custom method brings other defaulting to the parent method.
```

Comment:
```suggestion
Similarly, many developers have the reflex to write a custom `print.subclass()` method as part of the method implementation. While it may be justified in some cases, it is also sometimes unnecessary. My recommendation would be to evaluate carefully what benefit the custom method brings other defaulting to the parent method.
```